go/go121-release/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.21
```

Comment:
nit: I bet we could get by with just one version for releasing. The only reason we have two for CI is to test against lower and upper bound to make sure we are staying compatible for users.

Reply:
@codyoss can you confirm which one we should use?  Prior to me migrating release images there were discrepancies ex [google-api-go-client](https://source.corp.google.com/piper///depot/google3/cloud/sdk/client_libraries/kokoro/go/googleapis/google-api-go-client/release.cfg) used go123, while the rest under googleapis (gax-go, go-sql-spanner, etc) used go121

Reply:
I think 1.23, the latest version, would be fine.